CMakeLists.txt

```diff
@@ -1,5 +1,13 @@
 # Defines Taraxa library target.
 
+if (DEFINED SYSTEM_HOME_OVERRIDE)
+    list(APPEND CMAKE_PREFIX_PATH ${SYSTEM_HOME_OVERRIDE})
+    include_directories(${SYSTEM_HOME_OVERRIDE}/include)
+    link_directories(${SYSTEM_HOME_OVERRIDE}/lib)
+endif ()
+set(CMAKE_OSX_SYSROOT /)
```

Comment:
This one has effects only on mac, and it will tell cmake to use the system root as the toolchain root. Without this it will use some "default toolchain" in mac os, which doesn't have compilers, etc. installed by default, and god knows how to set it up properly.